src/mesh/index_derivs.cxx

```diff
@@ -1227,9 +1207,7 @@ const Field3D Mesh::indexDDY(const Field3D &f, CELL_LOC outloc,
       throw BoutException("Cannot use FFT for Y derivatives");
   }
 
-  result = applyYdiff(f, func, diffloc, region);
-
-  result.setLocation(diffloc); // Set the result location
+  Field3D result = applyYdiff(f, func, diffloc, region);
```

Comment:
Could just directly return `applyYdiff(f, func, diffloc, region);` I think.

Reply:
Yep, updated in ce76aa2.